tests/test_main.py

```diff
@@ -1,186 +1,91 @@
+
+
+def test_get_project(
+    mock_db: MagicMock, project_id: uuid.UUID, mock_project: models.Projects, mock_project_2: ProjectDetails
+) -> None:
+    mock_db.query.return_value.get.return_value = mock_project
+    result = get_project(project_id, mock_db)
+    assert ProjectDetails(
+        project_id=uuid.UUID(str(result.id)), name=str(result.name), description=str(result.description)
+    ) == ProjectDetails(
+        project_id=mock_project_2.project_id, name=mock_project_2.name, description=mock_project_2.description
+    )
 
 
-@pytest.fixture
-def created_projects(sample_project: Project, sample_project_2: Project) -> Generator[list[ProjectDetails]]:
-    from src.main import projects
-
-    for project in (sample_project, sample_project_2):
-        project_details = ProjectDetails(**project.model_dump())
-        projects.append(project_details)
-    yield projects
-
-
-def test_create_project(client: TestClient, sample_project: Project) -> None:
-    response = client.post("/projects", json=sample_project.model_dump())
-    assert response.status_code == 201
-    data = response.json()
-    expected_data = {
-        "name": sample_project.name,
-        "description": sample_project.description,
-        "project_id": data["project_id"],
-    }
-    assert data == expected_data
-
-
-def test_get_projects(client: TestClient, created_projects: list[ProjectDetails]) -> None:
-    response = client.get("/projects")
-    assert response.status_code == 200
-    data = response.json()
-    expected_data = [
-        {"name": project.name, "description": project.description, "project_id": str(project.project_id)}
-        for project in created_projects
+def test_get_projects(mock_db: MagicMock, mock_project: models.Projects, project_id: uuid.UUID) -> None:
+    mock_db.execute.return_value.scalars.return_value.all.return_value = [mock_project]
+    result: list[ProjectDetails] = get_projects_(mock_db)
+    expected_result = [
+        ProjectDetails(
+            project_id=uuid.UUID(str(mock_project.id)),
+            name=str(mock_project.name),
+            description=str(mock_project.description),
+        )
     ]
-    assert data == expected_data
+    assert result == expected_result
 
 
-def test_get_project_details(client: TestClient, created_project: ProjectDetails) -> None:
-    response = client.get(f"/project/{created_project.project_id}/info")
-    assert response.status_code == 200
-    data = response.json()
-    expected_data = {
-        "name": created_project.name,
-        "description": created_project.description,
-        "project_id": str(created_project.project_id),
-    }
-    assert data == expected_data
+def test_create_project(mock_db: MagicMock, project_data: Project) -> None:
+    models.Projects(id=uuid.uuid4(), name=project_data.name, description=project_data.description)
+    mock_db.add.return_value = None
+    mock_db.commit.return_value = None
+    mock_db.refresh.side_effect = lambda obj: obj
```

Comment:
you dont have to mock every action there, just mock a specific function which is there. And with mocked db you don't really test anything here. So I could suggest to remove this test. Check other similar db related tests and feel free to remove them.